### PR TITLE
Have ETCD Run as the Etcd User

### PR DESCRIPTION
**Pull request by briandowns**
This pull request sets up the ability for etcd in RKE2 to run as the etcd user by updating the necessary directories to being owned by the etcd user and adding a security context to the pod specification for etcd.

### CODE REVIEW

File: pkg/podexecutor/staticpod.go [hunk 1]
```diff
@@ -161,6 +164,17 @@ func (s *StaticPod) CurrentETCDOptions() (opts executor.InitialOptions, err erro
 	return
 }
 
+const (
+	basePath    = "/var/lib/rancher/rke2"
```
Comment (galal-hussein) [inline pkg/podexecutor/staticpod.go:168]: I dont think this should be hard coded to /var/lib/rancher/rke2, the datadir for rke2 can be changed with an option

File: pkg/podexecutor/staticpod.go [hunk 2]
```diff
@@ -197,5 +211,43 @@ func (s *StaticPod) ETCD(args executor.ETCDConfig) error {
 		HealthPort:  2381,
 		HealthPath:  "/health",
 		HealthProto: "HTTP",
+	}
+
+	if s.CISMode {
+		etcdUser, err := user.Lookup("etcd")
```
Comment (galal-hussein) [inline pkg/podexecutor/staticpod.go:217]: just a question here, should we consider creating the etcd user if it doesnt exist, or at least in the install script?

Comment (galal-hussein) [reply pkg/podexecutor/staticpod.go:217]: oh I see now that this mode has been added to the install script, nvm

File: pkg/podexecutor/staticpod.go (hunk 1 above)
Comment (briandowns) [reply pkg/podexecutor/staticpod.go:168]: Good point. I'll look to see how this can be avoided.

Comment (briandowns) [reply pkg/podexecutor/staticpod.go]: Resolved. Thanks!

**Pull request merged**